packages/cache/src/cache.ts

```diff
@@ -152,7 +158,7 @@ export async function saveCache(
   checkKey(key)
 
   const compressionMethod = await utils.getCompressionMethod()
-  let cacheId = null
+  let cacheId = -1
```

Comment:
why this change? IIRC this was made `null` recently. May @t-dedah can chime in on why it was made null.

Reply:
Okay, let me check with @t-dedah  because this function return type ``Number`` https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L150 so in case of exception returning null might not be correct

Reply:
Let me explain more why this change is required, earlier even though Function return type is number and ``cacheId`` was set null but in the code we were making sure this will never return null https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L204 otherwise we  were throwing error so this line https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L229 will never get execute (in case of exception). So compiler was happy. but now as i have added catch block https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L229 line can be executed (in case of no error or exception is consumed) so compiler is throwing error that ``return type can only be number not number | null

Reply:
Yep, that seems to be the case.